packages/expansion-hub/src/open-rev-hub.ts

```diff
@@ -10,12 +11,41 @@ import {ExpansionHubInternal} from "./internal/ExpansionHub";
  */
 const openSerialMap = new Map<string, Serial>();
 
+/**
+ * Opens a parent Rev Hub with a known address. Does not discover or open any child hubs.
+ * Call {@link ExpansionHub#addChildByAddress} to add known children.
+ *
+ * @param serialNumber
+ * @param moduleAddress
+ */
+export async function openParentExpansionHub(serialNumber: string, moduleAddress: number):
+    Promise<ParentRevHub & ExpansionHub> {
+    let serialPortPath = await getSerialPortPathForExHubSerial(serialNumber);
+
+    if(openSerialMap.get(serialPortPath) == undefined) {
+        openSerialMap.set(serialPortPath, await openSerial(serialPortPath));
+    }
+
+    let serial = openSerialMap.get(serialPortPath)!;
+
+    let parentHub = new ExpansionHubInternal(true, serialNumber);
+
+    await parentHub.open(serial, moduleAddress);
+    await parentHub.queryInterface("DEKA");
+
+    if(parentHub.isParent()) {
+        return parentHub;
+    } else {
+        throw new Error(`Hub at ${serialNumber} with moduleAddress ${moduleAddress} is not a parent`);
```

Comment:
I think we talked about some cases where we can make guarantees that as long as we don't make any mistakes, the consumer of the library won't have to worry about the case being possible. I usually include the checks anyways, since they make sure we don't make any mistakes (or at least throw a clearer message if so). I can just cast to ParentExpansionHub, but if we change the constructor to false (or have some sort of error inside of ExpansionHub), I don't think it can actually check that the cast should fail here, which could lead to odd behavior later. I generally prefer to convert undefined behavior to a clear Error whenever feasible.